src/mast/mod.rs

```diff
@@ -168,6 +171,54 @@ impl MastCtx {
     }
 }
 
+impl Symbolic {
+    /// Evaluate symbolic size to an integer.
+    pub fn eval(&self, typed_fn_env: &MonomorphizedFnEnv) -> u32 {
+        match self {
+            Symbolic::Concrete(v) => *v,
+            Symbolic::Generic(g) => typed_fn_env.get_type_info(&g.value).unwrap().value.unwrap(),
+            Symbolic::Add(a, b) => a.eval(typed_fn_env) + b.eval(typed_fn_env),
+            Symbolic::Mul(a, b) => a.eval(typed_fn_env) * b.eval(typed_fn_env),
+        }
+    }
+}
+
+impl GenericParameters {
```

Comment:
can you elaborate on this?

Reply:
usually you want to have

```rust
struct GenericParameters { /* ... */ }
impl GenericParameters { /* ... */ } // <-- the impl right after 
```

Reply:
I was trying to avoid importing `MonomorphizedFnEnv` in the `parser/types.rs`.
It is a bit weird the `parser/types.rs` depends on something relevant to monomorphization.

Reply:
yeah agree!